### PR TITLE
Refresh README

### PR DESCRIPTION
**Pull request by jin**
- Updated TOC and Features
- Made bzlmod instructions more prominent
- Moved linked to examples higher up in the README
- Upleveled BOM section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,57 +10,86 @@ Status](https://badge.buildkite.com/26d895f5525652e57915a607d0ecd3fc945c8280a0bd
...
+- [IPv6 support](#ipv6-support)
+- [Developing this project](#developing-this-project)
+    - [Verbose / debug mode](#verbose--debug-mode)
+    - [Tests](#tests)
+    - [Installing the Android SDK on macOS](#installing-the-android-sdk-on-macos)
+    - [Generating documentation](#generating-documentation)
 
 ## Features
 
+* MODULE.bazel bzlmod configuration (Bazel 6 and above only)
```
Comment (shs96c) [inline README.md:67]: We use Bazel 7-only features in our `MODULE.bazel`.

File: README.md [hunk 2]
```diff
@@ -10,57 +10,86 @@ Status](https://badge.buildkite.com/26d895f5525652e57915a607d0ecd3fc945c8280a0bd
...
 * WORKSPACE configuration
-* JAR, AAR, source JARs
+* Artifact version resolution with Coursier or Maven
+* Import downloaded JAR, AAR, source JARs
+* Export built JARs to Maven repositories
+* Pin resolved artifacts with their SHA-256 checksums into a version-controllable JSON file
 * Custom Maven repositories
-* Private Maven repositories with HTTP Basic Authentication
-* Artifact version resolution with Coursier
+* Private Maven repositories using HTTP Basic Authentication or `netrc`
```
Comment (shs96c) [inline README.md:74]: Do we want to encourage people to use basic authentication when Bazel itself won't? A simpler mental model of "if it works with Bazel, it'll work when resolving" might be better to lean into, and that suggests not mentioning basic auth....

File: README.md [hunk 3]
```diff
@@ -71,9 +100,19 @@ Support for Bazel versions before `4.0.0` is only available on rules_jvm_externa
 
 ## Usage
 
-> New: on Bazel 6, you can use bzlmod instead of following directions below.
-> See [bzlmod](./docs/bzlmod.md).
-> bzlmod is on-by-default in Bazel 7.0.
+### With bzlmod (Bazel 6 and above)
```
Comment (shs96c) [inline README.md:103]: Bazel 7.

**Review by shs96c — APPROVED** [review]
The bazel 6 -> 7 thing is important to fix before landing, but that's a minor edit and won't need another review :)  LGTM.

File: README.md (hunk 2 above)
Comment (jin) [reply README.md]: Fair point. Removed the basic auth mention.

**Pull request merged**